### ISSUE TITLE
[question] Can I deploy LLM on  SA8295 with Hexagon v68?

### ISSUE THREAD

**Issue by ecccccsgo** [quic/ai-hub-models#124]
hello guys, I'm trying to deploy LLM on SA8295 with NPU to accelerate inference. I try serval times but not success with the guidance `https://github.com/quic/ai-hub-apps/tree/main/tutorials/llm_on_genie` . but fail. I found `Only supports Hexagon v73 and onward architectures.` and the SA8295 support v68 on the doc `qairt/2.27.7.241014/docs/QNN/general/htp/htp_backend.html`. 

I hope to know if SA8295 not support the NPU inference  :(

Looking forward to your reply. Thank you.

**Comment by ecccccsgo** [comment]
i found the new chipset Snapdragon 8cx Gen 3 (SC8280X) with v68, and it seem that it support LLM with NPU in `https://docs.qualcomm.com/bundle/publicresource/topics/80-63442-50/overview.html`. Although it was PC chipset with msvc.

**Comment by kory** [comment]
Per the doc, we only support v73 architectures and above, so that device is unlikely to work.

**Comment by ecccccsgo** [comment]
> Per the doc, we only support v73 architectures and above, so that device is unlikely to work.

@kory thank you for reply. we found llama3.2 1b on v68 8295p in your website, hhhh. we will try it, if you got any update, hope you can share with us.

https://aihub.qualcomm.com/automotive/models/llama_v3_2_1b_instruct?chipsets=qualcomm-sa8295p

**Comment by Leonslam** [comment]
> > Per the doc, we only support v73 architectures and above, so that device is unlikely to work.
> 
> [@kory](https://github.com/kory) thank you for reply. we found llama3.2 1b on v68 8295p in your website, hhhh. we will try it, if you got any update, hope you can share with us.
> 
> https://aihub.qualcomm.com/automotive/models/llama_v3_2_1b_instruct?chipsets=qualcomm-sa8295p

Excuse me, but have you deploy it on 8295p?

**Comment by gustavla** [comment]
Yes, @kory misspoke. We do support v68, and we do support it for some LLMs (Llama 3.2 1B and 3B). 

You have to specify the precision w4 when exporting (--checkpoint DEFAULT_W4), unless that's already the default (1B). 

It should work, we have tested this on a real device (which is always the case when we present performance numbers on the AI Hub website).

Let us know if you are hitting any specific issues.

**Comment by Leonslam** [comment]
I try to deploy llama3.2 1b on v68 8295p, get errors like:

atlas:/data/local/tmp/llama/genie_bundle # ./genie-t2t-run -c genie_config.json -p "hello"
/genie-t2t-run -c genie_config.json -p "hello"                                <
Using libGenie.so version 1.11.0

[INFO]  "Using create From Binary"
[INFO]  "Allocated total size = 169165312 across 5 buffers"
[ERROR] "Could not create context from binary for context index = 0 : err 1002"
[ERROR] "Create From Binary FAILED!"
Failure to initialize model.
Failed to create the dialog.

@gustavla what does error 1002 mean? How to solve it? Looking forward to your reply. Thank you.

**Comment by metakong** [comment]
> I try to deploy llama3.2 1b on v68 8295p, get errors like:
> 
> atlas:/data/local/tmp/llama/genie_bundle # ./genie-t2t-run -c genie_config.json -p "hello" /genie-t2t-run -c genie_config.json -p "hello" < Using libGenie.so version 1.11.0
> 
> [INFO] "Using create From Binary" [INFO] "Allocated total size = 169165312 across 5 buffers" [ERROR] "Could not create context from binary for context index = 0 : err 1002" [ERROR] "Create From Binary FAILED!" Failure to initialize model. Failed to create the dialog.
> 
> [@gustavla](https://github.com/gustavla) what does error 1002 mean? How to solve it? Looking forward to your reply. Thank you.

You can get anything to work if you're willing to endure misery (Qwen on 8cx Gen 2 w/ nearly instant voice response via MS David and careful workarounds to decapitate Microsoft) = https://github.com/metakong/ARM64brute2020

**Comment by mestrona-3** [comment]
@metakong please file a new issue if you have a specific question for our team. Thanks.

**Comment by metakong** [comment]
> @metakong please file a new issue if you have a specific question for our team. Thanks. 

Not necessary.  Just wanted to give another user an example of how to get around the fact that neither Microsoft nor Qualcomm have any desire to support their products on 6 year old devices.  Thanks.
